lib/Myriad.pm

```diff
@@ -594,6 +596,20 @@ method setup_tracing () {
     return;
 }
 
+=head2 tell_parent
+
+Send a message to the parent process
+in case if Myriad is running under
+debug mode.
+
+=cut
+
+method tell_parent ($data) {
+    if ($parent_pipe) {
+        print $parent_pipe "$data\r\n";
```

Comment:
*Never, ever* use `\r\n` in Perl. It does not mean the same as C, the `\n` is mapped to whatever Perl feels like at the moment.

https://perldoc.perl.org/perlport#Newlines

Always use the `\x0D\x0A` sequence explicitly, or the `Socket.pm` `$CRLF` equivalent.

We should also never be using builtins when there are methods available, as a general style note - the exception is Myriad::Bootstrap which doesn't even have IO::Handle available.

Thirdly, as Torsten says this is unreliable. It should be an IO::Async::Stream, allowing you to write:

```suggestion
        $parent_pipe->write("$data\x0D\x0A");
```